lib/src/priority_queue.dart

```diff
@@ -223,18 +264,23 @@ class HeapPriorityQueue<E> implements PriorityQueue<E> {
   }
 
   @override
-  List<E> toList() =>
-      List<E>.generate(_length, (i) => _queue[i]!)..sort(comparison);
+  List<E> toList() => _toUnorderedList()..sort(comparison);
 
   @override
   Set<E> toSet() {
     var set = SplayTreeSet<E>(comparison);
     for (var i = 0; i < _length; i++) {
-      set.add(_queue[i]!);
+      set.add(_elementAt(i));
     }
     return set;
   }
 
+  @override
+  List<E> toUnorderedList() => _toUnorderedList();
```

Comment:
Why do we need the private one? Can we put the implementation here and skip the function?

Reply:
I call this method from `toList` too. It's an implementation detail that the two methods share code, so I don't want to risk someone overriding `toUnorderedList` and also affecting `toList`.
Generally, I avoid calling an overridable method from another method unless the latter method is explicitly documented as depending on the former. Code sharing is an implementation detail, so it shouldn't affect the visible API.